cmd/pipeline/deploy.go

```diff
@@ -281,12 +293,18 @@ func waitForResourcesToBeRunning(ctx context.Context, name, namespace string, ti
 					areAllRunning = false
 				}
 				if status == "error" {
-					errors[name] = 1
+					errorsMap[name] = 1
 				}
 			}
-			if len(errors) > 0 {
-				previewEnvURL := getNamespaceURL(ctx, name)
-				return fmt.Errorf("pipeline '%s' deployed with errors. You can check %s for more information", name, previewEnvURL)
+			if len(errorsMap) > 0 {
+				pipelineURL, err := getPipelineURL(ctx, name, namespace)
```

Comment:
could we store the pipeline object after calling the deploy API to avoid extra queries?